src/mentionui.js

```diff
@@ -0,0 +1,493 @@
+	 * @returns {TextWatcher}
+	 */
+	_setupTextWatcherForFeed( marker, minimumCharacters ) {
+		const editor = this.editor;
+
+		const watcher = new TextWatcher( editor, createTestCallback( marker, minimumCharacters ), createTextMatcher( marker ) );
+
+		watcher.on( 'matched', ( evt, data ) => {
+			const matched = data.matched;
+
+			const selection = editor.model.document.selection;
+
+			const hasMention = selection.hasAttribute( 'mention' );
+			const nodeBefore = selection.focus.nodeBefore;
+
+			if ( hasMention || nodeBefore && nodeBefore.is( 'text' ) && nodeBefore.hasAttribute( 'mention' ) ) {
+				return;
+			}
+
+			const { feedText, marker } = matched;
+
+			this._getFeed( marker, feedText )
+				.then( feed => {
+					this._items.clear();
+
+					for ( const name of feed ) {
+						const item = typeof name != 'object' ? { name } : name;
+
+						this._items.add( { item, marker } );
+					}
+
+					if ( this._items.length ) {
+						this._showPanel();
+					} else {
+						this._hidePanel();
+					}
+				} );
+		} );
+
+		watcher.on( 'unmatched', () => {
+			this._hidePanel();
+		} );
+
+		return watcher;
+	}
+
+	/**
+	 * Returns registered text watcher for marker.
+	 *
+	 * @private
+	 * @param {String} marker
+	 * @returns {TextWatcher}
+	 */
+	_getWatcher( marker ) {
+		const { watcher } = this._mentionsConfigurations.get( marker );
+
+		return watcher;
+	}
+
+	/**
+	 * Shows the {@link #panelView}. If panel is already visible it will reposition it.
+	 *
+	 * @private
+	 */
+	_showPanel() {
+		this.panelView.pin( this._getBalloonPanelPositionData() );
+		this.panelView.show();
+		this._mentionsView.selectFirst();
+	}
+
+	/**
+	 * Hides the {@link #panelView}.
+	 *
+	 * @private
+	 */
+	_hidePanel() {
+		this.panelView.unpin();
+		this.panelView.hide();
+	}
+
+	/**
+	 * Renders the single item in autocomplete list.
+	 *
+	 * @private
+	 * @param {module:mention/mention~MentionFeedItem} item
+	 * @param {String} marker
+	 * @returns {module:ui/button/buttonview~ButtonView|module:mention/ui/domwrapperview~DomWrapperView}
+	 */
+	_renderItem( item, marker ) {
+		const editor = this.editor;
+
+		let view;
+
+		const renderer = this._getItemRenderer( marker );
+
+		if ( renderer ) {
+			const domNode = renderer( item );
+
+			view = new DomWrapperView( editor.locale, domNode );
+		} else {
+			const buttonView = new ButtonView( editor.locale );
+
+			buttonView.label = item.name;
+			buttonView.withText = true;
+
+			view = buttonView;
+		}
+
+		return view;
+	}
+
+	/**
+	 * @returns {module:utils/dom/position~Options}
+	 * @private
+	 */
+	_getBalloonPanelPositionData() {
+		const view = this.editor.editing.view;
+		const domConverter = view.domConverter;
+		const viewSelection = view.document.selection;
+
+		return {
+			target: () => {
+				const range = viewSelection.getLastRange();
+				const rangeRects = Rect.getDomRangeRects( domConverter.viewRangeToDom( range ) );
+
+				return rangeRects[ rangeRects.length - 1 ];
```

Comment:
`pop()`

Reply:
Anyway, when testing, let's keep an eye on Safari, because it had [problems obtaining rects for ranges in the past](https://github.com/ckeditor/ckeditor5-link/issues/46).